packages/clerk-js/src/ui/components/SignIn/SignInStart.tsx

```diff
@@ -64,7 +64,17 @@ export function _SignInStart(): JSX.Element {
     placeholder: localizationKeys('formFieldInputPlaceholder__password') as any,
   });
 
-  const identifierField = useFormControl('identifier', '', {
+  const initialValues: Record<SignInStartIdentifier, string | undefined> = useMemo(
+    () => ({
+      email_address: ctx.initialValues?.emailAddress,
+      email_address_username: ctx.initialValues?.emailAddress || ctx.initialValues?.username,
+      username: ctx.initialValues?.username,
+      phone_number: ctx.initialValues?.phoneNumber,
+    }),
+    [JSON.stringify(ctx.initialValues)],
```

Comment:
Why did u find stringifying necessary ? Shouldn't `ctx.initialValues` get a new reference when one of its properties is updated ?